### ISSUE TITLE
Flowering Oak leaves render weirdly in "Fast" graphics mode

### ISSUE THREAD

**Issue by BBoldt** [Glitchfiend/BiomesOPlenty#696]
**Fancy:** 
![](http://i.imgur.com/YU4CLeg.jpg)
**Fast:** 
![](http://i.imgur.com/6aom87o.png)

BOP 1922
Forge 1761

**Comment by Zygus42** [comment]
Also BoP 2052 and many different versions of Forge.

**Comment by BBoldt** [comment]
Still an issue in BOP 2082, Forge 2076
![](http://i.imgur.com/OFkfQfY.jpg)

**Comment by Forstride** [comment]
well, yes, it was never fixed, so...

the only solution would either be to have the flowering leaves render normally, or have the flowers colored like leaves.  which would you prefer?

**Comment by BBoldt** [comment]
Just thought I would do an update with a newer version since the original issue was on 1.8.9.

Flowers colored like leaves would probably be best, that way they would still be distinct. Anything would be better than the solid white cubes. I've been noticing them a lot lately since 1.10 Modded Minecraft is killing my fps and forcing me to lower graphics settings.

**Comment by Divineaspect** [comment]
In a similar boat, mostly commenting to keep apprised of this issue's resolution.

**Comment by Adubbz** [comment]
Flowering leaves now render similar to oak leaves in fast mode, with flowers coloured like leaves (unintentionally as suggested by BBolt :P). There's not really any way of getting proper layering to work on fast graphics as far as I can tell, since rendering is such a pain in the ass nowadays. 

This is what it looks like for reference:
http://i.imgur.com/TGjN1pe.jpg

**Comment by xAlicatt** [comment]
This doesn't appear to be fixed in version 2109 for 1.10.2 Please advise.

![2016-12-22_16 48 11](https://cloud.githubusercontent.com/assets/24724298/21442942/d6543236-c866-11e6-9929-bad964bafb88.png)

**Comment by Adubbz** [comment]
Because it wasn't fixed in that version? Only 2117 and newer contain the fix. If someone wishes to backport it to 1.10 then they're welcome to submit a pull request.

**Comment by xAlicatt** [comment]
Understood. Is this something that could be "fixed" with a texture change perhaps?

**Comment by Adubbz** [comment]
That would work, although it would also mean the fancy model would be messed up also. Unfortunately this was something that required additional code to achieve

**Comment by xAlicatt** [comment]
Is there any chance you would implement texture change in the 1.10 version that would make the block render as fast leaves regardless of whether fast or fancy graphics are chosen? I'd venture to say that a large percentage or maybe even most people use fast graphics on modded minecraft since it tends to stress systems much more than vanilla does. 

With all due respect, I'm not sure why development on the 1.10 version has stopped, since it seems to be the version with the most mods available among modpacks greater than 1.8. I looked into just using 1.11 for my pack I'm working on and a significant portion of the mods aren't at 1.11 right now..not even a beta release. I feel like 1.10 is becoming and will be a popular modpack version for a long time to come based on what I'm seeing in available mods.

Thank you for your time :)

**Comment by Adubbz** [comment]
But as I mentioned earlier, that's just swapping one issue for another really. I'd need to see the stats to back up the 'large percentage' you mention, because anecdotally I've seen the opposite.

As for 1.10 support, there's a few factors here. First, backports are something we've tried to avoid at all costs - It's highly demotivating and time consuming implementing the exact same code again and again with masses of name changes and other adaptations having to be made for an old version. If another mod developer wishes to make the changes and submit a pull request, that's fine, but it's not something we're going to be supporting ourselves.

We moved to 1.11 as soon as possible simply because it's the latest version available. I understand not all mods may take that approach, and I respect that decision, but I ask that you respect ours also. Supporting older versions denies us access to newer features in both Minecraft and Forge that we may find useful, and ultimately just results in another 1.7.10 scenario where we sit on the same version for around 2 years (despite Minecraft receiving numerous updates since then), and those who do try to move forward are bombarded with hatred and pestering.

Finally, I really don't want the development of BoP to be affected by the decisions made by other modders for other mods. BoP has already been removed from some major modpacks such as FTB (for reasons that were rather questionable/laughable, but that's a story for another day), and going forward I wouldn't be surprised if more follow. Mods these days appear to be highly replacable, and that's perfectly fine, but my priority lies with ensuring the longevity of BoP. Though there may be "better" alternatives, I hope to do my best to ensure BoP always remains an option available. With that said, compatibility with other mods isn't something I'm going to go out of my way to prevent, and I'll try to fix issues when I get around to it, but honestly for me at least, it's not what is driving development forward.

At the end of the day, it's not a matter of being difficult for the sake of being difficult. It's moreso trying to prolong my own interest in updating the mod, whilst also sticking to the beliefs both Forstride and I hold about BoP's development and avoiding whatever other chaos may exist in the modding community at any given time.
